diff --git a/src/ingest_subgraph.rs b/src/ingest_subgraph.rs
index fbead94..d8f355e 100644
--- a/src/ingest_subgraph.rs
+++ b/src/ingest_subgraph.rs
@@ -10,7 +10,8 @@ mod nested_key_fields;
 mod schema_definitions;
 
 use self::{
-    directive_definitions::*, directives::*, nested_key_fields::ingest_nested_key_fields, schema_definitions::*,
+    directive_definitions::*, directives::*, nested_key_fields::ingest_nested_key_fields,
+    schema_definitions::*,
 };
 use crate::{
     Subgraphs,
@@ -59,12 +60,11 @@ fn ingest_top_level_definitions(ctx: &mut Context<'_>) {
 
     for definition in ctx.document.definitions() {
         match definition {
-            ast::Definition::Type(type_definition) | ast::Definition::TypeExtension(type_definition) => {
+            ast::Definition::Type(type_definition)
+            | ast::Definition::TypeExtension(type_definition) => {
                 let type_name = type_definition.name();
 
-                let description = type_definition
-                    .description()
-                    .map(|description| ctx.subgraphs.strings.intern(description.to_cow()));
+                let description = None;
 
                 let definition_id = match type_definition {
                     ast::TypeDefinition::Object(_) if type_name == SERVICE_TYPE_NAME => continue,
@@ -86,7 +86,8 @@ fn ingest_top_level_definitions(ctx: &mut Context<'_>) {
                                 ctx.subgraphs.set_mutation_type(subgraph_id, definition_id);
                             }
                             RootTypeMatch::Subscription => {
-                                ctx.subgraphs.set_subscription_type(subgraph_id, definition_id);
+                                ctx.subgraphs
+                                    .set_subscription_type(subgraph_id, definition_id);
                             }
                             RootTypeMatch::NotRootButHasDefaultRootName => {
                                 ctx.subgraphs.push_ingestion_diagnostic(subgraph_id, format!("The {type_name} type has the default name for a root but is itself not a root. This is not valid in a federation context."));
@@ -96,16 +97,20 @@ fn ingest_top_level_definitions(ctx: &mut Context<'_>) {
 
                         definition_id
                     }
-                    ast::TypeDefinition::Interface(_interface_type) => ctx.subgraphs.get_or_push_definition(
+                    ast::TypeDefinition::Interface(_interface_type) => {
+                        ctx.subgraphs.get_or_push_definition(
+                            subgraph_id,
+                            type_name,
+                            DefinitionKind::Interface,
+                            description,
+                        )
+                    }
+                    ast::TypeDefinition::Union(_) => ctx.subgraphs.get_or_push_definition(
                         subgraph_id,
                         type_name,
-                        DefinitionKind::Interface,
+                        DefinitionKind::Union,
                         description,
                     ),
-                    ast::TypeDefinition::Union(_) => {
-                        ctx.subgraphs
-                            .get_or_push_definition(subgraph_id, type_name, DefinitionKind::Union, description)
-                    }
                     ast::TypeDefinition::InputObject(_) => ctx.subgraphs.get_or_push_definition(
                         subgraph_id,
                         type_name,
@@ -120,20 +125,27 @@ fn ingest_top_level_definitions(ctx: &mut Context<'_>) {
                         description,
                     ),
 
-                    ast::TypeDefinition::Enum(_enum_type) => {
-                        ctx.subgraphs
-                            .get_or_push_definition(subgraph_id, type_name, DefinitionKind::Enum, description)
-                    }
+                    ast::TypeDefinition::Enum(_enum_type) => ctx.subgraphs.get_or_push_definition(
+                        subgraph_id,
+                        type_name,
+                        DefinitionKind::Enum,
+                        description,
+                    ),
                 };
 
                 let directive_site_id = ctx.subgraphs.at(definition_id).directives;
-                directives::ingest_directives(ctx, directive_site_id, type_definition.directives(), |_| {
-                    type_name.to_owned()
-                });
+                directives::ingest_directives(
+                    ctx,
+                    directive_site_id,
+                    type_definition.directives(),
+                    |_| type_name.to_owned(),
+                );
 
                 directives::ingest_keys(definition_id, type_definition.directives(), ctx);
             }
-            ast::Definition::Directive(_) | ast::Definition::Schema(_) | ast::Definition::SchemaExtension(_) => (),
+            ast::Definition::Directive(_)
+            | ast::Definition::Schema(_)
+            | ast::Definition::SchemaExtension(_) => (),
         }
     }
 }
@@ -151,7 +163,10 @@ fn ingest_definition_bodies(ctx: &mut Context<'_>) {
         match definition {
             ast::TypeDefinition::Union(_) if definition.name() == ENTITY_UNION_NAME => continue,
             ast::TypeDefinition::Union(union) => {
-                let Some(union_id) = ctx.subgraphs.definition_by_name(definition.name(), subgraph_id) else {
+                let Some(union_id) = ctx
+                    .subgraphs
+                    .definition_by_name(definition.name(), subgraph_id)
+                else {
                     ctx.subgraphs.push_ingestion_diagnostic(
                         subgraph_id,
                         format!(
@@ -163,7 +178,9 @@ fn ingest_definition_bodies(ctx: &mut Context<'_>) {
                 };
 
                 for member in union.members() {
-                    let Some(member_id) = ctx.subgraphs.definition_by_name(member.name(), subgraph_id) else {
+                    let Some(member_id) =
+                        ctx.subgraphs.definition_by_name(member.name(), subgraph_id)
+                    else {
                         ctx.subgraphs.push_ingestion_diagnostic(
                             subgraph_id,
                             format!(
@@ -177,7 +194,10 @@ fn ingest_definition_bodies(ctx: &mut Context<'_>) {
                 }
             }
             ast::TypeDefinition::InputObject(input_object) => {
-                let Some(definition_id) = ctx.subgraphs.definition_by_name(definition.name(), subgraph_id) else {
+                let Some(definition_id) = ctx
+                    .subgraphs
+                    .definition_by_name(definition.name(), subgraph_id)
+                else {
                     ctx.subgraphs.push_ingestion_diagnostic(
                         subgraph_id,
                         format!(
@@ -190,7 +210,10 @@ fn ingest_definition_bodies(ctx: &mut Context<'_>) {
                 fields::ingest_input_fields(ctx, definition_id, input_object.fields());
             }
             ast::TypeDefinition::Interface(interface) => {
-                let Some(definition_id) = ctx.subgraphs.definition_by_name(interface.name(), subgraph_id) else {
+                let Some(definition_id) = ctx
+                    .subgraphs
+                    .definition_by_name(interface.name(), subgraph_id)
+                else {
                     ctx.subgraphs.push_ingestion_diagnostic(
                         subgraph_id,
                         format!(
@@ -202,8 +225,9 @@ fn ingest_definition_bodies(ctx: &mut Context<'_>) {
                 };
 
                 for implemented_interface in interface.implements_interfaces() {
-                    let Some(implemented_interface) =
-                        ctx.subgraphs.definition_by_name(implemented_interface, subgraph_id)
+                    let Some(implemented_interface) = ctx
+                        .subgraphs
+                        .definition_by_name(implemented_interface, subgraph_id)
                     else {
                         ctx.subgraphs.push_ingestion_diagnostic(
                             subgraph_id,
@@ -217,7 +241,8 @@ fn ingest_definition_bodies(ctx: &mut Context<'_>) {
                         continue;
                     };
 
-                    ctx.subgraphs.push_interface_impl(definition_id, implemented_interface);
+                    ctx.subgraphs
+                        .push_interface_impl(definition_id, implemented_interface);
                 }
 
                 let is_query_root_type = false; // interfaces can't be at the root
@@ -226,7 +251,10 @@ fn ingest_definition_bodies(ctx: &mut Context<'_>) {
             }
             ast::TypeDefinition::Object(_) if definition.name() == SERVICE_TYPE_NAME => continue,
             ast::TypeDefinition::Object(object_type) => {
-                let Some(definition_id) = ctx.subgraphs.definition_by_name(definition.name(), subgraph_id) else {
+                let Some(definition_id) = ctx
+                    .subgraphs
+                    .definition_by_name(definition.name(), subgraph_id)
+                else {
                     ctx.subgraphs.push_ingestion_diagnostic(
                         subgraph_id,
                         format!(
@@ -238,8 +266,9 @@ fn ingest_definition_bodies(ctx: &mut Context<'_>) {
                 };
 
                 for implemented_interface in object_type.implements_interfaces() {
-                    let Some(implemented_interface) =
-                        ctx.subgraphs.definition_by_name(implemented_interface, subgraph_id)
+                    let Some(implemented_interface) = ctx
+                        .subgraphs
+                        .definition_by_name(implemented_interface, subgraph_id)
                     else {
                         ctx.subgraphs.push_ingestion_diagnostic(
                             subgraph_id,
@@ -253,7 +282,8 @@ fn ingest_definition_bodies(ctx: &mut Context<'_>) {
                         continue;
                     };
 
-                    ctx.subgraphs.push_interface_impl(definition_id, implemented_interface);
+                    ctx.subgraphs
+                        .push_interface_impl(definition_id, implemented_interface);
                 }
 
                 let is_query_root_type = ctx.root_type_matcher.is_query(definition.name());
@@ -261,7 +291,10 @@ fn ingest_definition_bodies(ctx: &mut Context<'_>) {
                 fields::ingest_fields(ctx, definition_id, object_type.fields(), is_query_root_type);
             }
             ast::TypeDefinition::Enum(enum_definition) => {
-                let Some(enum_id) = ctx.subgraphs.definition_by_name(definition.name(), subgraph_id) else {
+                let Some(enum_id) = ctx
+                    .subgraphs
+                    .definition_by_name(definition.name(), subgraph_id)
+                else {
                     ctx.subgraphs.push_ingestion_diagnostic(
                         subgraph_id,
                         format!(
@@ -279,7 +312,10 @@ fn ingest_definition_bodies(ctx: &mut Context<'_>) {
     }
 }
 
-pub(super) fn ast_value_to_subgraph_value(value: ConstValue<'_>, subgraphs: &mut Subgraphs) -> subgraphs::Value {
+pub(super) fn ast_value_to_subgraph_value(
+    value: ConstValue<'_>,
+    subgraphs: &mut Subgraphs,
+) -> subgraphs::Value {
     match &value {
         ConstValue::Null(_) => subgraphs::Value::Null,
         ConstValue::Int(n) => subgraphs::Value::Int(n.as_i64()),
@@ -287,9 +323,11 @@ pub(super) fn ast_value_to_subgraph_value(value: ConstValue<'_>, subgraphs: &mut
         ConstValue::String(s) => subgraphs::Value::String(subgraphs.strings.intern(s.as_str())),
         ConstValue::Boolean(b) => subgraphs::Value::Boolean(b.value()),
         ConstValue::Enum(e) => subgraphs::Value::Enum(subgraphs.strings.intern(e.name())),
-        ConstValue::List(l) => {
-            subgraphs::Value::List(l.items().map(|v| ast_value_to_subgraph_value(v, subgraphs)).collect())
-        }
+        ConstValue::List(l) => subgraphs::Value::List(
+            l.items()
+                .map(|v| ast_value_to_subgraph_value(v, subgraphs))
+                .collect(),
+        ),
         ConstValue::Object(o) => subgraphs::Value::Object(
             o.fields()
                 .map(|field| {
diff --git a/src/ingest_subgraph/fields.rs b/src/ingest_subgraph/fields.rs
index 920168c..072745d 100644
--- a/src/ingest_subgraph/fields.rs
+++ b/src/ingest_subgraph/fields.rs
@@ -17,9 +17,7 @@ pub(super) fn ingest_input_fields(
             format!("{}.{field_name}", subgraphs[parent_definition.name])
         });
 
-        let description = field
-            .description()
-            .map(|description| ctx.subgraphs.strings.intern(description.to_cow()));
+        let description = None;
 
         let default = field
             .default_value()
@@ -53,27 +51,30 @@ fn ingest_field_arguments(
         ingest_directives(ctx, directives, argument.directives(), |subgraphs| {
             let field_name = &subgraphs[parent_field_name];
             let parent_definition_name = &subgraphs[subgraphs.at(parent_definition_id).name];
-            format!("{}.{}({}:)", parent_definition_name, field_name, argument.name())
+            format!(
+                "{}.{}({}:)",
+                parent_definition_name,
+                field_name,
+                argument.name()
+            )
         });
 
-        let description = argument
-            .description()
-            .as_ref()
-            .map(|description| ctx.subgraphs.strings.intern(description.to_cow()));
+        let description = None;
 
         let default_value = argument
             .default_value()
             .map(|default| ast_value_to_subgraph_value(default, ctx.subgraphs));
 
-        ctx.subgraphs.insert_field_argument(subgraphs::ArgumentRecord {
-            parent_definition_id,
-            parent_field_name,
-            name,
-            r#type,
-            directives,
-            description,
-            default_value,
-        });
+        ctx.subgraphs
+            .insert_field_argument(subgraphs::ArgumentRecord {
+                parent_definition_id,
+                parent_field_name,
+                name,
+                r#type,
+                directives,
+                description,
+                default_value,
+            });
     }
 }
 
@@ -91,9 +92,7 @@ pub(super) fn ingest_fields(
             continue;
         }
 
-        let description = field
-            .description()
-            .map(|description| ctx.subgraphs.strings.intern(description.to_cow()));
+        let description = None;
 
         let field_type = ctx.subgraphs.intern_field_type(field.ty());
         let directives = ctx.subgraphs.new_directive_site();
